platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/security/session/AttributeSharingHashSessionIdManager.java

```diff
@@ -83,37 +78,26 @@
 
     @Override
     public void invalidateSession(
-        String subjectName, Function<HttpSession, String> sessionSubjectExtractor) {
+        String subjectName, Function<Map<String, Object>, String> sessionSubjectExtractor) {
 
-      final List<AbstractSession> matchingSessions =
-          getHttpSessionStream()
-              .filter(s -> subjectName.equals(sessionSubjectExtractor.apply(s)))
-              .filter(AbstractSession.class::isInstance)
-              .map(AbstractSession.class::cast)
-              .distinct()
-              .collect(Collectors.toList());
-      for (AbstractSession matchingSession : matchingSessions) {
-        idManager.removeSession(matchingSession);
-        if (matchingSession.isValid()) {
-          matchingSession.invalidate();
-        }
-      }
-    }
+      final Optional<String> sessionIdOptional =
+          getSessionAttributeMapsStream()
+              .filter(e -> subjectName.equals(sessionSubjectExtractor.apply(e.getValue())))
+              .map(e -> e.getKey())
+              .findFirst();
```

Comment:
No. Before we were holding a mapping of sessionId -> set of sessions (across different contexts). This isn't needed for the purposes of sharing attributes with this new architecture. Also, when we call `invalidate` on the session, it will notify the session handler of all contexts to invalidate their sessions.